Core.ApplicationServices/References/ReferenceService.cs

```diff
@@ -200,22 +200,43 @@ private void RaiseRootUpdated(IEntityWithExternalReferences owner)
                             case > 1:
                                 return new OperationError("Only one reference can be master reference", OperationFailure.BadInput);
                         }
-
-                        foreach (var referenceProperties in newReferences)
+                        
+                        var referencesToUpdate = new List<ExternalReference>();
+                        foreach (var externalReferenceProperties in referenceList)
                         {
-                            var result = AddReference(rootId, rootType, referenceProperties.Title, referenceProperties.DocumentId, referenceProperties.Url);
-
-                            if (result.Failed)
-                                return new OperationError($"Failed to add reference with data:{JsonConvert.SerializeObject(referenceProperties)}. Error:{result.Error.Message.GetValueOrEmptyString()}", result.Error.FailureType);
+                            ExternalReference externalReference;
+                            if (externalReferenceProperties.Uuid.HasValue)
```

Comment:
a one line comment explaining the cause of the different blocks might make sense